client/my-sites/checkout/checkout-main-wrapper.tsx

```diff
@@ -80,6 +81,7 @@ export default function CheckoutMainWrapper( {
 	const translate = useTranslate();
 	const locale = useSelector( getCurrentUserLocale );
 	const selectedSiteId = useSelector( getSelectedSiteId ) ?? undefined;
+	useGoalsFirstExperiment();
```

Comment:
We need this to ensure the experiment assignment is stored in memory and we can retrieve it to render the feature list. But I don't love that it implicitly assigns the user to a cohort if they haven't been assigned yet.

Who knows where `checkout-main-wrapper` is being included from, causing an assignment to occur.

I think the hook's internal `flow === 'onboarding'` check will be enough to save us though.